### PR TITLE
Trig constant fold

### PR DESCRIPTION
**Pull request by ajpal**

### CODE REVIEW

File: src/language.rs [hunk 1]
```diff
@@ -39,15 +40,42 @@ impl<L: SynthLanguage> Analysis<L> for SynthAnalysis {
...
+            rec
+        } else {
+            let mut nodes: Vec<L> = vec![];
+            let mut map: HashMap<Id, Id> = HashMap::default();
+            enode.for_each(|id| {
+                if map.get(&id).is_none() {
+                    let s = get_simplest(&id);
+                    let i = nodes.len();
+                    for n in s.as_ref() {
+                        nodes.push(n.clone().map_children(|id| Id::from(usize::from(id) + i)));
```
Comment (chandrakananandi) [inline src/language.rs:57]: not following what's happening here. The `Id::from(usize::from(id) + i)` part specifically.

**Review by chandrakananandi — APPROVED** [review]

File: src/language.rs (hunk 1 above)
Comment (chandrakananandi) [reply src/language.rs:57]: actually @bksaiki you did this I think originally? what was this for?

Comment (bksaiki) [reply src/language.rs:57]: Hmmm... good question

Comment (bksaiki) [reply src/language.rs:57]: I think it's just constructing a RecExpr, but I don't remember why

**Pull request merged**